intro_to_ada.rst

```diff
@@ -1904,23 +1904,747 @@ Abstract data types
 Generics
 ========
 
-Generic declaration
--------------------
-
-Generic body
+Introduction
 ------------
 
+Generics are used for metaprogramming in Ada. They are useful for abstract
+algorithms that share common properties.
+
+Generics can be used for subprograms or packages. A generic is declared
+by using the following format:
+
+.. code-block:: ada
+
+    generic
```

Comment:
I would rather you follow the style of the rest of the course in this section, which is:

1. Show a complete, syntactically correct, stupid example of the feature
2. Explain syntax stuff in the comments
3. Explain how it works in the following paragraph

Here your examples are not syntactically correct, and you cut a single example into several parts. It will be a bother for the automatic code runner, is in my opinion confusing for the user, and is not coherent with the rest of the course.